graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -567,9 +717,44 @@ class RenameSchemaTypesVisitor(Visitor):
     # named "Foo", suppressed_type_names will contain "Foo".
     suppressed_type_names: Set[str]
 
+    # reverse_field_name_map maps type name to a dict, which in turn maps the name of a field in the
+    # renamed schema to the name of the field in the original schema, if the field has different
+    # names in the original schema and the new schema. If field_renamings would rename a field named
+    # "foo" (in a type named "Baz") to "bar", then reverse_field_name_map["Baz"] will map "bar" to
+    # "foo".
+    reverse_field_name_map: DefaultDict[str, Dict[str, str]]
+
+    # Collects no-op renamings for fields, mapping the type name that contains the field to the set
+    # of field names for which field_renamings contained no-op renamings. Applies only to types for
+    # which field_renamings.get(type_name, set()) is iterable. For a type named "Bar", if
+    # field_renaming
+    #    - renames a field named "foo" to "foo", or
+    #    - attempts to rename a field named "foo" when such a field does not exist
+    # no_op_field_renamings will map "Bar" to a set containing "foo".
+    no_op_field_renamings: DefaultDict[str, Set[str]]
+
+    # Collects type names for each object type that has field renamings that have been applied.
+    # After every renaming is done, this is used to ensure that field_renamings contains no unused
+    # field renamings for a particular type if field_renamings is iterable.
+    types_with_field_renamings_processed: Set[str]
+
+    # Collects invalid field names in field_renamings. If field_renamings would rename a field named
+    # "foo" (in a type named "Bar") to a string that is not a valid, non-reserved GraphQL type name
+    # (valid, non-reserved names consist only of alphanumeric characters and underscores, do not
+    # start with a number, and do not start with two underscores), invalid_field_names will map
+    # "Bar" to a dict that maps "foo" to the invalid field name.
+    invalid_field_names: DefaultDict[str, Dict[str, str]]
+
+    # Collects naming conflict errors involving fields. If field_renamings would rename multiple
+    # fields (in a type named "Bar" in the original schema) to "foo", field_name_conflicts will map
+    # "Bar" to a dict that maps "foo" to a set containing the names of the fields in the original
+    # schema that would be renamed to "foo".
+    field_name_conflicts: Dict[str, Dict[str, Set[str]]]
```

Comment:
Would welcome suggestions on how to better organize the data structures in here, since this visitor class has picked up quite a few fields just because of the various types
of errors that might arise during renaming and needing to keep track of it all.